src/libAtomVM/globalcontext.c

```diff
@@ -102,6 +102,40 @@ Context *globalcontext_get_process(GlobalContext *glb, int32_t process_id)
     return NULL;
 }
 
+typedef void *(*context_iterator)(Context *ctx, void *accum);
+
+static void * increment_context_count(Context *ctx, void *accum)
+{
+    return (void *) ((size_t) accum + 1);
+}
+
+static void *cons_context(Context *ctx, void *accum)
+{
+    return (void *) term_list_prepend(term_from_local_process_id(ctx->process_id), (term) accum, ctx);
+}
+
+static void *globalcontext_iterate_processes(GlobalContext *glb, context_iterator fun, void *accum)
+{
+    Context *processes = GET_LIST_ENTRY(glb->processes_table, Context, processes_table_head);
+    Context *p = processes;
+    do {
+        accum = fun(p, accum);
+        p = GET_LIST_ENTRY(p->processes_table_head.next, Context, processes_table_head);
+    } while (processes != p);
+    return accum;
+}
+
+size_t globalcontext_num_processes(GlobalContext *glb)
+{
+    return (size_t) globalcontext_iterate_processes(glb, increment_context_count, NULL);
+}
+
+term globalcontext_list_processes(GlobalContext *glb)
```

Comment:
this function will be unlikely used in other places, so it would be better to keep it static in nifs.c